Develop

ChadRehmKineticData opened the pull request:
Updated the org repo create handler for use with clone template workflow process

Commit:
Updated create org repo handler

Added a success output and error message handling.

Commit:
Added to results xml for org repo create handler
Added Handle Error Message and Response Body

Commit:
added private option for create repo

Commit:
updated node.xml format for org repo create handler

Commit:
Added new github api handler.

ChadRehmKineticData committed:
Merge pull request #2 from kineticdata/feature/generic-handler

Added new github api handler.

jackboespflug commented on handlers/github_api_v1/handler/init.rb:
```diff
@@ -0,0 +1,98 @@
+    rescue RestClient::Exception => e
+      response_code = e.response.code
+
+      puts "There was an error parsing the JSON error response" if @debug_logging_enabled
```
This error message is wrong. You are rescuing a RestClient exception, not a JSON Parse exception.

ChadRehmKineticData replied:
Good catch, thanks

ChadRehmKineticData replied:
Changed message to be accurate.

jackboespflug commented on handlers/github_organization_repo_create_v1/changelog.md:
```diff
@@ -0,0 +1,5 @@
+* Initial version.  See README for details.
+
+Github_Repo_Create_Organizaiton_V1 (2020-10-13)
+* Added success output and error message handling.
```
Misspellings: Github_Repo_Create_Organizaiton_V1 should be Github_Repo_Create_Organization_V1.

I didn't see the success output addition.

Since you are changing the info_values/parameters, should the entire handler be changed to v2?

ChadRehmKineticData replied:
I discuss a version change with James and Ben.  It was decided that since this is only adding to info_values and parameter and it is not a heavily used handler that we would keep it at v1.  I'll correct the readme.

jackboespflug replied:
OK as long as you guys talked about it.

ChadRehmKineticData replied:
Removed "success" so that the language better reflected the changes to the handler.

jackboespflug commented on handlers/github_organization_repo_create_v1/handler/init.rb:
```diff
@@ -20,22 +20,47 @@ def initialize(input)
       </results>
       RESULTS
+    rescue RestClient::Exception => e
+      error_message = e.inspect
```
Why `e.inspect` instead of just `e.message`?

ChadRehmKineticData replied:
I have no specific preference for either.  It is a holdover from where I got the original code.  Does .message have the same result as .inspect?

ChadRehmKineticData replied:
Jack, I looked into this more.  The [Ruby Exception doc](https://www.rubydoc.info/gems/rest-client/1.6.1/RestClient/Exception) show that inspect returns "#{message}: #{http_body}".  I think I like that more than just returning the message.

jackboespflug replied:
OK

jackboespflug commented on handlers/github_organization_repo_create_v1/readme.md:
```diff
@@ -1,13 +1,13 @@
-[access_token]
+### Info
+* [access_token]
   As of 2/26/1016, Access Tokens using GitHub's oauth2 in an organization requires approval than a callbackURL, where the access token is sent to.
```
This wasn't part of your changes, but the date referenced for access_token is off by a 1000 years. Maybe just fix that with your changes.

ChadRehmKineticData replied:
Sorry Jack,  I changed the for format of the readme.md file.  It was originally just README without the md extension and the file didn't conform to markdown spec.  I changed it so that it would look nicer when viewed from the github UI.  The date was a hold over from the original code base.  I'll get it corrected in the next version.

ChadRehmKineticData replied:
corrected date

jackboespflug commented on changelog.md:
```diff
@@ -0,0 +1,5 @@
+Github [handlers] 
+* [github_organization_repo_create_v1] (2020-10-13)
+    * Added success output and error message handling.
```
I didn't see the `success` output addition.

ChadRehmKineticData replied:
It wasn't called "success" the output was actually called "output".  output is only populated when the request response code is 200.  I can remove "success" to make it clearer what was added.

ChadRehmKineticData replied:
Removed success to better reflect the changes to the handler.

jackboespflug replied:
I don't follow this logic. What is the point of only adding the result 'output' when the response code is 200?

jackboespflug reviewed (changes requested):
@ChadRehmKineticData Please see my comments.

Commit:
Made updates based on code review
Change log language was made clearer and typos corrected.  Error message
in github_api handler was correct to be accurate.  Date in readme was correct.

jackboespflug commented on handlers/github_api_v1/readme.md:
```diff
@@ -0,0 +1,33 @@
+## Results
+[Response Body]
+  The returned value from the Rest Call (JSON format)
+
```
I missed this in my original review - but you are missing the "Response Code" and "Handler Error Message" results.

jackboespflug commented on handlers/github_api_v1/readme.md:
```diff
@@ -0,0 +1,33 @@
+  select the scopes 'repo' and the appropriate corresponding subsequent scopes you desire in the repository creation process.
+  Since you need a callback URL, the user may want to consider a third party web client such as Google's PostMan in
+  order to make the request and generate the token.  Once the user has acquired this token, they may reuse it.
+  * A personal [access token](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token) will also work.
```
I missed this in my original review - you are missing the "enable_debug_logging" info value.

jackboespflug commented on handlers/github_organization_repo_create_v1/handler/init.rb:
```diff
@@ -20,22 +20,47 @@ def initialize(input)
       results = <<-RESULTS
       <results>
+        <result name="Response Body">#{escape(response.nil? ? {} : response.body)}</result>
+        <result name="Handler Error Message">#{escape(error_message)}</result>
```
All of the results should always be returned in case they are referenced in the process tree by connectors or other nodes. You can store the values in variables, and then build one results block after the begin/rescue block. Take a look how you did this is the generic Github API handler.

jackboespflug commented on handlers/github_organization_repo_create_v1/readme.md:
```diff
@@ -1,20 +1,20 @@
-== GitHub Organization Repository Creation
+# GitHub Organization Repository Creation
```
Should be "GitHub Organization Repository Create" for consistency.

jackboespflug commented on handlers/github_organization_repo_create_v1/readme.md:
```diff
@@ -1,20 +1,20 @@
+    * Name of desired repository user wishes to create.
+  * [org_name]
+    * Name of organization of which you wish to create a new repository in.
 
```
Missed "enable_debug_logging" parameter.

jackboespflug commented on handlers/github_organization_repo_create_v1/readme.md:
```diff
@@ -1,20 +1,20 @@
   Since you need a callback URL, the user may want to consider a third party web client such as Google's PostMan in
   order to make the request and generate the token.  Once the user has acquired this token, they may reuse it.
 
 
```
Missing "org_name" and "enable_debug_logging" info values.

Commit:
Added org_name as a parameter
If org_name is provided as a parameter will override the info_value.

Commit:
updated changelog

Commit:
updated readme styles

Commit:
added missing label to node.xml and added parameters to readme.

Commit:
updated readme

Commit:
updated readme and info value description
